### PR TITLE
#rotate_vectors!

### PR DESCRIPTION
**Pull request by JulieT1008**

### CODE REVIEW

**Comment by JulieT1008** [comment]
@thomasnaude

File: lib/daru_lite/dataframe.rb [hunk 1]
```diff
@@ -1157,6 +1157,27 @@ def order=(order_array)
...
+    #     total: [5, 7, 9],
+    #   })
+    #   df.rotate_vectors(-1)
+    #   df
+    #   # => #<DaruLite::DataFrame(3x3)>
+    #   #       total b   a
+    #   #   0   5     4   1
+    #   #   1   7     5   2
+    #   #   2   9     6   3
+    def rotate_vectors!(count = -1)
```
Comment (thomasnaude) [inline lib/daru_lite/dataframe.rb]: remove the bang in the method name as it is used in the current codebase only for destructive methods that have a non-destructive alternative or for methods returning self

File: lib/daru_lite/dataframe.rb [hunk 2]
```diff
@@ -1157,6 +1157,27 @@ def order=(order_array)
       initialize(to_h, order: order_array)
     end
 
+    # Rotate the vectors in a dataframe
+    # @param count => Integer, param for the rotate array method
```
Comment (thomasnaude) [inline lib/daru_lite/dataframe.rb]: It is not clear how that params is working.
Use Ruby's documentation as an example of how to explain the role of the count parameter

**Review by thomasnaude — CHANGES_REQUESTED** [review]

File: lib/daru_lite/dataframe.rb [hunk 3]
```diff
@@ -1157,6 +1157,27 @@ def order=(order_array)
       initialize(to_h, order: order_array)
     end
 
+    # Rotate the vectors in a dataframe
```
Comment (thomasnaude) [inline lib/daru_lite/dataframe.rb]: add an indication to what is returned by the method

File: lib/daru_lite/dataframe.rb [hunk 4]
```diff
@@ -1157,6 +1157,27 @@ def order=(order_array)
...
+    #   df
+    #   # => #<DaruLite::DataFrame(3x3)>
+    #   #       total b   a
+    #   #   0   5     4   1
+    #   #   1   7     5   2
+    #   #   2   9     6   3
+    def rotate_vectors!(count = -1)
+      return self unless vectors.many?
+
+      self.order = vectors.to_a.rotate(count)
```
Comment (thomasnaude) [inline lib/daru_lite/dataframe.rb:1180]: that method is returning different objects based on the number of vectors

File: spec/dataframe_spec.rb [hunk 5]
```diff
@@ -4274,4 +4274,22 @@ def create_test(*args, &proc)
...
+        b: [4,5,6],
+        total: [5,7,9]
+      })
+    end
+    let(:new_order) { [:total, :a, :b]}
+
+    it "changes the position of the last vector to first one" do
+      subject
+      expect(df.vectors.to_a).to eq(new_order)
+    end
```
Comment (thomasnaude) [inline spec/dataframe_spec.rb:4302]: add a test for when there is only one vector

**Review by thomasnaude — CHANGES_REQUESTED** [review]

**Comment by JulieT1008** [comment]
@thomasnaude

**Review by thomasnaude — APPROVED** [review]

**Pull request merged**